cli/dcoscli/tables.py

```diff
@@ -849,10 +850,31 @@ def slave_table(slaves):
         ('ID', lambda s: s['id'])
     ])
 
-    tb = table(fields, slaves, sortby="HOSTNAME")
+    for field_name in field_names:
+        if field_name.lower() == 'ip':
+            fields[field_name.upper()] = lambda s: mesos.parse_pid(s['pid'])[1]
```

Comment:
since `hostname`, `ip`, and `id` will always be included, if they are specified with `--field` we shouldn't add them to the table again.